src/TodoApp.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {
+  FormEvent,
+  useEffect,
+  useState,
+} from 'react';
+import { useLocation } from 'react-router-dom';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { FilterTypes } from './types/FilterTypes';
+import { Notification } from './components/Notification';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(() => {
+    const localTodos = localStorage.getItem('todos');
+
+    return localTodos ? JSON.parse(localTodos) : [];
+  });
+
+  const [title, setTitle] = useState('');
+  const [notification, setNotification] = useState<string>('');
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  const filteredTodos = todos.filter(({ completed }) => {
+    switch (pathname) {
+      case FilterTypes.COMPLETED:
+        return completed;
+
+      case FilterTypes.ACTIVE:
+        return !completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const addNewTodo = (event: FormEvent) => {
+    event.preventDefault();
+    setNotification('');
+    const newTitle = title.trim();
+
+    if (!newTitle || !/\S/.test(newTitle)) {
+      setNotification('Title can\'t be empty');
+
+      return;
+    }
+
+    const newId = +new Date();
+
+    const todoToAdd = {
+      id: newId,
+      title: newTitle,
+      completed: false,
+    };
+
+    setTodos(previousTodos => [...previousTodos, todoToAdd]);
+    setTitle('');
+  };
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const switchTodoStatus = (id: number, status: boolean) => {
+    setTodos(
+      todos.map((todo: Todo) => {
+        return {
+          ...todo,
+          ...(id === todo.id && { completed: status }),
+        };
+      }),
```

Comment:
```suggestion
      todos.map((todo: Todo) => ({
        ...todo,
        ...(id === todo.id && { completed: status }),
      }),
```
You can also just immediately return an object